Packs/Traceable/Integrations/Traceable/Traceable.py

```diff
@@ -882,32 +1026,51 @@ def fetch_incidents(client: Client, last_run, first_fetch_time):
 
     latest_created_time: datetime = _last_fetch
     incidents = []
-    items = client.get_threat_events(_last_fetch, datetime.now())
+    items = client.get_threat_events(_last_fetch, datetime.utcnow())
     demisto.info(f"Retrieved {len(items)} records.")
     if len(items) > 0:
         demisto.debug(f"First Incident: {json.dumps(items[0], indent=3)}")
     for item in items:
+        context_value = None
+        context_key = ""
+        if client.fetch_unique_incidents:
+            context_key = (f"{item['environment']}_{item['serviceName']}_{item['name']}_{item['apiId']}_"
+                           + f"{item['anomalousAttribute']}")
+            context_value = client.get_integration_context_key_value(context_key)
+            if context_value is not None:
+                context_entry_date = Helper.string_to_datetime(context_value)
+                if context_entry_date < datetime.utcnow():
+                    context_value = None
         incident_created_time: datetime = datetime.fromtimestamp(
             item["timestamp"]["value"] / 1000
         )
-        incident = {
-            "name": item["name"],
-            "displayname": item["displayname"],
-            "country": item["country"],
-            "sourceip": item["sourceip"],
-            "riskscore": item["riskscore"],
-            "ipAddressType": item["ipAddressType"],
-            "severity": XSOAR_SEVERITY_BY_TRACEABLE_SEVERITY.get(
-                item["severity"], IncidentSeverity.UNKNOWN
-            ),
-            "rawJSON": json.dumps(item),
-        }
-        if ("eventUrl" in item
-                and item["eventUrl"] is not None
-                and item["eventUrl"] != ""):
-            incident["eventUrl"] = item["eventUrl"]
-
-        incidents.append(incident)
+        if context_value is None:
+            demisto.info(f"Context key {context_key} not found in instance cache. Creating a new incident.")
+            incident = {
+                "name": item["name"],
+                "displayname": item["displayname"],
+                "country": item["country"],
+                "sourceip": item["sourceip"],
+                "riskscore": item["riskscore"],
+                "ipAddressType": item["ipAddressType"],
+                "severity": XSOAR_SEVERITY_BY_TRACEABLE_SEVERITY.get(
+                    item["severity"], IncidentSeverity.UNKNOWN
+                ),
+                "rawJSON": json.dumps(item),
+            }
+            if ("eventUrl" in item
+                    and item["eventUrl"] is not None
+                    and item["eventUrl"] != ""):
```

Comment:
Please use the dict `get` method
https://www.w3schools.com/python/ref_dictionary_get.asp

Reply:
Done